submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/service/TimesService.kt

```diff
@@ -14,16 +18,38 @@ import java.time.ZoneOffset.UTC
 /**
  * Calculates the submission release date based on current state of the submission in the system. Calculation rules.
  */
-class TimesService {
+class TimesService(
+    private val privilegesService: IUserPrivilegesService,
+) {
     internal fun getTimes(request: SubmitRequest): Times {
         val now = OffsetDateTime.now()
         val creationTime = request.previousVersion?.creationTime ?: now
-        val releaseTime = request.submission.releaseDate?.let { parseDate(it) }
+        val releaseTime = request.submission.releaseDate?.let { releaseTime(now, it, request) }
         val released = releaseTime?.isBeforeOrEqual(now).orFalse()
 
         return Times(creationTime, now, releaseTime, released)
     }
 
+    private fun releaseTime(
+        now: OffsetDateTime,
+        releaseDate: String,
+        request: SubmitRequest,
+    ): OffsetDateTime {
+        val releaseTime = parseDate(releaseDate)
+        val accNo = request.accNo
+        val user = request.submitter.email
+        val today = now.toInstant().asOffsetAtStartOfDay()
```

Comment:
lets said is 2pm accotrding to this if we try to release to same day at 8 am (in the past) code still will fail which does not look correct.

Reply:
The exact time isn't taken into consideration for the release date since it's [currently parsed](https://github.com/EBIBioStudies/biostudies-backend-services/blob/master/submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/service/TimesService.kt#L30) always at the start of the day [regardless if it contains the time](https://github.com/EBIBioStudies/biostudies-backend-services/blob/master/submission/submitter/src/test/kotlin/ac/uk/ebi/biostd/submission/service/TimesServiceTest.kt#L74)

Reply:
make sense to me